torchvision/prototype/transforms/functional/_augment.py

```diff
@@ -1,52 +1,68 @@
 from typing import Tuple
+from typing import TypeVar
 
 import torch
-from torchvision.transforms import functional as _F
+from torchvision.prototype import features
+from torchvision.prototype.transforms import kernels as K
 
+from ._utils import dispatch
 
-erase_image = _F.erase
+T = TypeVar("T", bound=features.Feature)
 
 
-def _mixup(input: torch.Tensor, batch_dim: int, lam: float, inplace: bool) -> torch.Tensor:
-    if not inplace:
-        input = input.clone()
+@dispatch(
+    {
+        features.Image: K.erase_image,
+    },
+)
+def erase(input: T, *, i: int, j: int, h: int, w: int, v: torch.Tensor, inplace: bool) -> T:
+    """ADDME"""
+    ...
 
-    input_rolled = input.roll(1, batch_dim)
-    return input.mul_(lam).add_(input_rolled.mul_(1 - lam))
 
+@dispatch(
+    {
+        features.Image: K.mixup_image,
+        features.OneHotLabel: K.mixup_one_hot_label,
+    },
+)
+def mixup(input: T, *, lam: float, inplace: bool) -> T:
+    """ADDME"""
+    ...
 
-def mixup_image(image_batch: torch.Tensor, *, lam: float, inplace: bool = False) -> torch.Tensor:
-    if image_batch.ndim < 4:
-        raise ValueError("Need a batch of images")
 
-    return _mixup(image_batch, -4, lam, inplace)
+@dispatch(
+    {
+        features.Image: K.cutmix_image,
+        features.OneHotLabel: K.cutmix_one_hot_label,
+    },
+)
+def cutmix(input: T, *, box: Tuple[int, int, int, int], lam_adjusted: float, inplace: bool) -> T:
+    """Perform the CutMix operation as introduced in the paper
```

Comment:
Rendered documentation looks like this

![Screenshot from 2022-02-09 22-57-43](https://user-images.githubusercontent.com/6849766/153297237-748d6ae8-1a9d-456d-8c9a-a8de09a3c26d.png)

I'm not super happy with the design yet, but it has all the information that I think is necessary on the dispatcher. Of course the kernels documentation is clickable.

Reply:
How does the documentation look like now that you switched to `*args: Any, **kwargs: Any`. I'm concerned that as a user I don't know what parameters I need to pass. Also if I need to go to the low-level kernel, am I guaranteed that the two will match exactly in terms of parameters? I think there were exceptions, right?